service.go

```diff
@@ -108,7 +110,9 @@ func (s *Service) ServeHTTPInContext(c Context, w http.ResponseWriter, r *http.R
 		}
 
 		if handler == nil {
-			http.NotFoundHandler().ServeHTTP(w, r)
+			if s.notFound != nil {
+				s.notFound(c, w, r, func() {})
+			}
```

Comment:
I think we should still call `http.NotFoundHandler().ServeHTTP(w, r)` in an `else` clause here.